src/handlers/psp.ts

```diff
@@ -200,90 +200,58 @@ const initializeEvmStates = async (state: State): Promise<State> => {
-        async () => {
-          try {
-            return await callApi(config, endpoint, apiCallParameters);
-          } catch (err) {
-            if (hasApiCallTimedOut) {
-              return [
-                [utils.logger.pend('ERROR', err instanceof Error ? err.message : String(err))],
-                { templateId: template.id, apiValue: null, subscriptions },
-              ] as CallApiResult;
-            }
-            throw err;
-          }
-        },
-        { timeoutMs: 10_000, retries: 0 }
-      );
-      if (result.success) {
-        const [logs, data] = result.data;
-        return [logs, { templateId: template.id, apiValue: data, subscriptions }] as CallApiResult;
-      } else {
-        errorLogs.push(
-          utils.logger.pend('DEBUG', `Retrying API call for templateId ${template.id}: ${result.error.message}`)
-        );
-        hasApiCallTimedOut = result.error.message.includes('Operation timed out');
-        throw result.error;
-      }
+    const infiniteRetries = 100_000;
+    const goResult = await promise.go(() => callApi(config, endpoint, apiCallParameters), {
+      attemptTimeoutMs: 10_000,
+      retries: infiniteRetries,
+      totalTimeoutMs: 40_000,
+      delay: {
+        type: 'random',
+        minDelayMs: 200,
+        maxDelayMs: 2000,
+      },
     });
-    responses.push(result);
-  });
-
-  const result = await promise.go(async () => await Promise.all(apiValuePromises), { timeoutMs: 40_000, retries: 0 });
-  utils.logger.logPending(errorLogs, baseLogOptions);
-  if (!result.success) {
-    hasApiCallWrapperTimedOut = result.error.message.includes('Operation timed out');
-    utils.logger.error(`An error has occurred while calling APIs: ${result.error.message}`, baseLogOptions);
-  }
 
-  const apiValuesBySubscriptionId = responses.reduce((acc: { [subscriptionId: string]: ethers.BigNumber }, result) => {
-    const [logs, data] = result;
+    if (goResult.success) {
+      const [logs, data] = goResult.data;
+      return [logs, { templateId: template.id, apiValue: data, subscriptions }] as CallApiResult;
+    } else {
+      return [
+        [utils.logger.pend('DEBUG', `Retrying API call for templateId ${template.id}: ${goResult.error.message}`)],
+        { templateId: template.id, apiValue: null, subscriptions },
+      ] as CallApiResult;
+    }
+  });
 
-    const templateLogOptions = buildLogOptions('additional', { templateId: data.templateId }, baseLogOptions);
+  const callApiResults = await Promise.all(apiValuePromises);
+  const successfullCalls = callApiResults.filter((call) => call[1].apiValue !== null);
+  const failedCalls = callApiResults.filter((call) => call[1].apiValue === null);
 
-    utils.logger.logPending(logs, templateLogOptions);
+  utils.logger.logPending(
+    successfullCalls.flatMap((call) => call[0]),
+    baseLogOptions
+  );
+  utils.logger.logPending(
+    failedCalls.flatMap((call) => call[0]),
+    baseLogOptions
+  );
```

Comment:
I understand why we filter successful calls, we use those in `reduce`. But why do we need the failed ones? You're logging them the same way as the successful ones anyway. You can filter the successful ones and log all of them.